feat: Rename pybigquery to sqlalchemy-bigquery

jimfulton opened the pull request:
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [x] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/python-bigquery-sqlalchemy/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [x] Ensure the tests and linter pass
- [x] Code coverage does not decrease (if any source code was changed)
- [x] Appropriate docs were updated (if necessary)

Fixes #197  🦕

Commit:
Rename refs in files

Commit:
Move package

Commit:
more renaming

Commit:
Expose dialect and types

Commit:
explain the weird registration

Commit:
fixed: No longer need `test_pybigquery_sqla` schema, we generte random schema names.

Commit:
document that we obsolete pybigquery

Commit:
get _type_map from the base module.

It, itself isn't public.

Commit:
fix import

Commit:
Warn if an old (non-empty) version of pybigquery is installed.

Commit:
Add __version__

Commit:
fixed names

Commit:
Updated the way the path to __init__.py is computed

and updated package name in copyright statement.

Commit:
Updated copyright statements with new package name

Commit:
Define the version in version.py

Commit:
skip coverage for code handling pybigquery hanging around

tswast commented:
Could we add a warning to setup.py and the README and make a release of pybigquery first? I'd like to make sure folk who look for the pybigquery package are able to find sqlalchemy-bigquery.

I'm thinking something similar to what we're doing here: https://pypi.org/project/google-cloud/

jimfulton commented:
> Could we add a warning to setup.py and the README and make a release of pybigquery first? I'd like to make sure folk who look for the pybigquery package are able to find sqlalchemy-bigquery.
> 
> I'm thinking something similar to what we're doing here: https://pypi.org/project/google-cloud/

In this case, we'd be telling people to install a package that doesn't exist, wouldn't we?

Note that pybigquery will continue to work indefinitely!

Daft plan, modified based on your comment:

- Release an alpha of sqlalchemy-bigquery
- Release an alpha of pybigquery that is a facade/shim for sqlalchemy-bigquery
  Non-alpha pybigquery works as expected.
  
  Prior to these releases, test with non-pypi installs.
  
- Decide when to go to non-alpha.  For sake of discussion, say, July 42. ;)
- Make a pybigquery release that gives folks a heads up of the swap on July 42.
- On July 42, release non-alpha sqlalchemy-bigquery and non-alpha pybigquery shim
  
  pybigquery shim still works but issues a deprecation warning.

  Old installs of pybigquery or installes of pinned versions still work as expected.

Thoughts?

Commit:
blacken/lint

Commit:
Merge branch 'master' into sqlalchemy-bigquery-197

Commit:
merge upstream

Commit:
Merge remote-tracking branch 'origin/sqlalchemy-bigquery-197' into sqlalchemy-bigquery-197

Commit:
merge upstream

tswast commented:
> In this case, we'd be telling people to install a package that doesn't exist, wouldn't we?

True. Maybe we try to use some of the LTS client automation and keep around a `pybigquery` branch for backports? Or maybe we just manually publish a release of `pybigquery` once `sqlalchemy-bigquery` is available?

jimfulton commented:
There are a number of ways we could approach this, depending on what we
(you :) ) want to do. Let's discuss on Wednesday.

Jim

On Tue, Jul 13, 2021 at 3:17 PM Tim Swast ***@***.***> wrote:

> In this case, we'd be telling people to install a package that doesn't
> exist, wouldn't we?
>
> True. Maybe we try to use some of the LTS client automation and keep
> around a pybigquery branch for backports? Or maybe we just manually
> publish a release of pybigquery once sqlalchemy-bigquery is available?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/googleapis/python-bigquery-sqlalchemy/pull/198#issuecomment-879336432>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAF4U5YTOHRYENEPHAZOINTTXSGI3ANCNFSM47W3FT6A>
> .
>


-- 
Jim Fulton
http://jimfulton.info

tswast commented on .repo-metadata.json:
```diff
@@ -1,10 +1,10 @@
+  "name": "sqlalchemy-bigquery",
   "name_pretty": "SQLAlchemy dialect for BigQuery",
-  "client_documentation": "https://github.com/googleapis/pybigquery",
+  "client_documentation": "https://github.com/googleapis/sqlalchemy-bigquery",
```
```suggestion
  "client_documentation": "https://googleapis.dev/python/sqlalchemy-bigquery/latest/index.html",
```

jimfulton committed:
Update .repo-metadata.json

Co-authored-by: Tim Swast <swast@google.com>

Commit:
Merge branch 'master' into sqlalchemy-bigquery-197

jimfulton commented:
Note that we have to sort out the new name, since the new name was taken.

Commit:
Merge branch 'master' into sqlalchemy-bigquery-197

cpdean commented:
> Note that we have to sort out the new name, since the new name was taken.

Hey @jimfulton and @tswast , I have added Jim as an owner on https://pypi.org/project/sqlalchemy_bigquery/ , and added a redirect notice on my old project at https://github.com/cpdean/sqlalchemy-bigquery .

Let me know if there's anything else I can do to help.

jimfulton commented:
> > Note that we have to sort out the new name, since the new name was taken.
> 
> Hey @jimfulton and @tswast , I have added Jim as an owner on https://pypi.org/project/sqlalchemy_bigquery/ , and added a redirect notice on my old project at https://github.com/cpdean/sqlalchemy-bigquery .
> 
> Let me know if there's anything else I can do to help.

@cpdean Thank you very much!

Commit:
merge upstream/master

jimfulton commented:
@tswast are you good with this?

Commit:
merge upstream/master

Commit:
Keep pybigquery history pypi link

tswast commented on .repo-metadata.json:
```diff
@@ -1,11 +1,12 @@
   "library_type": "INTEGRATION",
   "repo": "googleapis/python-bigquery-sqlalchemy",
-  "distribution_name": "pybigquery",
+  "distribution_name": "sqlalchemy-bigquery"
```
Need extra comma now that we added `api_id`

```suggestion
  "distribution_name": "sqlalchemy-bigquery",
```

jimfulton replied:
Fixed

tswast commented on sqlalchemy_bigquery/__init__.py:
```diff
@@ -0,0 +1,73 @@
+except ImportError:
+    pass
+else:  # pragma: NO COVER
+    if not hasattr(pybigquery, "__version__"):
```
I'm curious what the `not hasattr` is doing. Could you add a comment about this? If `__version__` _is_ present it's less likely to conflict?

jimfulton replied:
That's a holdover from the shim, I think.  The shim would have that attribute.  I'll get rid of it.

jimfulton replied:
fixed

tswast reviewed (commented):
Looking good! A couple minor comments. I think we can merge this sooner rather than later.

Commit:
Removed the api module

Commit:
added missing comma

Commit:
we decided not to have a shim, so get rid of the check for it.

Commit:
fixed indentation

jimfulton commented:
> Looking good! A couple minor comments. I think we can merge this sooner rather than later.

I also just removed the `api` module, which we'd discussed, but that I just remembered. :)

tswast commented on sqlalchemy_bigquery/__init__.py:
```diff
@@ -16,22 +16,55 @@
+except ImportError:
+    pass
+else:  # pragma: NO COVER
+    warnings.warn(
```
Missing import for `warnings`

jimfulton replied:
fixed

tswast commented on sqlalchemy_bigquery/__init__.py:
```diff
@@ -16,22 +16,55 @@
-        assert job_config.dry_run
-        assert not job_config.use_query_cache
+try:
+    import pybigquery
```
Looks like we need to tell `flake8` to ignore this line.

jimfulton replied:
fixed

Commit:
Added missing import and added module docstring

Commit:
More docs reorg

Commit:
fixed some heading levels and also highlight that older versions are distributed as pybigquery

Commit:
lint

tswast commented on README.rst:
```diff
@@ -98,7 +98,7 @@ API Client
 
 .. code-block:: python
 
-    from pybigquery.api import ApiClient
+    from sqlalchemy_bigquery.api import ApiClient
     api_client = ApiClient()
     print(api_client.dry_run_query(query=sqlstr).total_bytes_processed)
```
We can remove this code sample now that we don't have an `ApiClient`

jimfulton replied:
Dang, I did yesterday, but I guess that change lost in my docfx flailing.  Thanks, will do.

jimfulton replied:
done

Commit:
removed a reference to the old API client
